src/Entities/Player.cpp

```diff
@@ -941,10 +941,32 @@ void cPlayer::ApplyArmorDamage(int DamageBlocked)
 	{
 		ArmorDamage = 1;
 	}
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 0, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 1, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 2, ArmorDamage);
-	m_Inventory.DamageItem(cInventory::invArmorOffset + 3, ArmorDamage);
+
+	for (int i = 0; i < 4; i++)
+	{
+		// (60 + (40 / (UnbreakingLevel + 1)))% chance the armor takes damage
+		// Ref: https://minecraft.gamepedia.com/Enchanting#Unbreaking
+		cItem ArmorPiece = m_Inventory.GetArmorSlot(i);
+		if (ArmorPiece.IsEmpty())
+		{
+			continue;
+		}
+		unsigned int UnbreakingLevel = ArmorPiece.m_Enchantments.GetLevel(cEnchantments::enchUnbreaking);
+		double chance = (0.6 + (0.4 / (UnbreakingLevel + 1)));
+
+		// When durability is reduced by multiple points
+		// Unbreaking is applied for each point of reduction.
+		std::binomial_distribution<short> Dist(ArmorDamage, chance);
+		auto ReducedDamage = Dist(GetRandomProvider().Engine());
```

Comment:
(I'll personally prefer to use the concrete types for simple values. E.g. `short` instead of `auto`.)